lib/models/api_calls.dart

```diff
@@ -98,5 +110,108 @@ class ApiCalls {
     }
   }
 
-  // TODO Remote commands Issue #38
+  /// Calls the Directory API to deprovise a device.
+  ///
+  /// * [authToken] is the authentication token for API to accept the requests.
+  /// * [deviceId] is the ID of the device which its detailed information
+  /// will be retrieved.
+  /// * [reason] is the reason why the device is getting a deprovision.
+  ///
+  /// This method returns true if the action was completed.
+  dynamic deprovision(String authToken, String deviceId, int reason) async {
```

Comment:
nit: Is it possible to use an enum for the radial box? And passs that instead?

Reply:
Done, I pass the index of the enum.

Reply:
You could make your reasons be a mapping indexed by an actual boolean rather than an int.  If the int is bounded it would make sense to use an enum so you know the value is not invalid.

If you don't do this you should check for invalid values here too, right now I think you will get an index out of range or something if the index is not right.

Reply:
Done, I did a map with the enum and the strings.